app/services/api/declarations/void.rb

```diff
@@ -0,0 +1,75 @@
+module API::Declarations
+  class Void
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    VOIDABLE_PAYMENT_STATUSES = %w[no_payment eligible payable ineligible].freeze
+
+    attribute :declaration_id
+    attribute :voided_by_user_id
+
+    validate :not_voided
+    validate :voidable, unless: :clawing_back?
+    validate :not_already_refunded, if: :clawing_back?
+    validate :output_fee_statement_available, if: :clawing_back?
+
+    def clawback_or_void
+      return false unless valid?
+
+      if clawing_back?
+        Declarations::Void.new.clawback
+      else
+        Declarations::Void.new.void
+      end
+    end
+
+  private
+
+    def declaration
+      @declaration ||= Declaration.find_by!(api_id: declaration_id)
+    end
+
+    def clawing_back? = declaration.payment_status_paid?
+
+    def not_voided
+      return if errors[:declaration_id].any?
+      return unless declaration.payment_status_voided?
+
+      errors.add(:declaration_id, "The declaration has already been voided")
+    end
+
+    def voidable
+      return if errors[:declaration_id].any?
+      return if declaration.payment_status.in?(VOIDABLE_PAYMENT_STATUSES)
```

Comment:
If we move the constant to `Declaration` we could have `declaration.voidable_payment?`